frameworks/FSharp/giraffe/src/App/Program.fs

```diff
@@ -128,20 +130,25 @@ module Main =
                 NewtonsoftJson.Serializer(NewtonsoftJson.Serializer.DefaultSettings)
                 :> Json.ISerializer
 
-        WebHostBuilder()
-            .UseKestrel()
-            .Configure(
-                fun builder ->
-                    builder
-                        .UseRouting()
-                        .UseGiraffe HttpHandlers.endpoints |> ignore)
-            .ConfigureServices(
-                fun services ->
-                    services
-                        .AddRouting()
-                        .AddGiraffe()
-                        .AddSingleton(jsonSerializer)
-                    |> ignore)
-            .Build()
-            .Run()
+        let configureApp (appBuilder : IApplicationBuilder) =

```

Comment:
Glancing at this, is there anything here that could cause a performance change? UseKestrel being dropped? Anything else?

Reply:
AFAIK that's how hosting is initialized in asp.net core 6. /cc @davidfowl

Reply:
This could be a bit more minimal overall:

```F#
let builder = WebApplication.CreateBuilder(args)

builder.Services.AddGiraffe()
                .AddSingleton(jsonSerializer) |> ignore

let app = builder.Build()

app.UseGiraffe HttpHandlers.endpoints |> ignore

app.Run()
```

Reply:
There is another PR coming that disables logging. Could this be a change to asp.net core 5 that affected performance? Was logging enabled before?

Reply:
And thanks for the help!

Reply:
@davidfowl for the avoidance of doubt, is that because these things are all registered by default:
https://github.com/dotnet/aspnetcore/blob/main/src/DefaultBuilder/src/WebApplicationBuilder.cs#L238-L299

Reply:
@davidfowl AddRouting could indeed go but removing UseRouting errors with:

```
Unhandled exception. System.InvalidOperationException: EndpointRoutingMiddleware matches endpoints setup by EndpointMiddleware and so must be added to the request execution pipeline before EndpointMiddleware. Please add EndpointRoutingMiddleware by calling 'IApplicationBuilder.UseRouting' inside the call to 'Configure(...)' in the application startup code.
```

Are we expected to add this inside the UseGiraffe implementation?

Reply:
Change this to MapGiraffeEndpoints instead of UseGiraffe

cc @dustinmoris

Reply:
When the `EndpointRouting` namespace is open then the `UseGiraffe` extension method is [doing the correct thing](https://github.com/giraffe-fsharp/Giraffe/blob/master/src/Giraffe/EndpointRouting.fs#L366-L376) already.